File: Angular-material/angular-covalent/src/app/service/location.service.ts
import { Location } from './../model/Location';
import { PageRequest } from './../model/PageRequest';

import { Observable } from 'rxjs/Observable';
import { Http, Headers, Response } from '@angular/http';
import { Injectable } from '@angular/core';

/**
 * 
 */
@Injectable()
export class LocationService 
{
 
  /*-------------------------------------------------------------------
	 * 		 					ATTRIBUTES
	 *-------------------------------------------------------------------*/
  /**
   * 
   */
  headers: Headers;
  /**
   * 
   */
  url: String = '/projeto/location/';


  /*-------------------------------------------------------------------
	 * 		 					CONSTRUCTOR
	 *-------------------------------------------------------------------*/
  constructor(public http: Http) 
  { 
    this.headers = new Headers();
    this.headers.append('Content-Type', 'application/json');
  }

  /*-------------------------------------------------------------------
	 *				 		     SERVICES
	 *-------------------------------------------------------------------*/
  /**
   * 
   * @param location 
   */
  insertLocation(location): Observable<Response>
  {
    if(location.id != undefined )
    {
       return this.http.put(this.url + 'updateLocation', JSON.stringify(location), { headers: this.headers });
    }
    else
    {
       return this.http.post(this.url + 'insertLocation', JSON.stringify(location), { headers: this.headers });
    }
  }
 
  /**
   * 
   */
  listAllLocation(): Observable<Location[]>
  {
    return this.http.get(this.url + 'listAllLocation').map(res => res.json());
  }
  
  /**
   * 
   * @param id 
   */
  listAllSubLocation(id): Observable<Location[]>
  {
    return this.http.get(this.url + 'listAllSubLocation/' + id).map(res => res.json());
  }
   /**
    * 
    */
  deleteLocation(location): Observable<Response>
  {
    return this.http.delete(this.url + 'deleteLocation/' + location.id);
  }

  /**
   * 
   * @param id 
   */
  findLocationbyId(id): Observable<Location>
  {
    return this.http.get(this.url + 'findLocationById/' + id).map(res => res.json());;
  }

  /**
   * 
   * @param page 
   * @param size 
   * @param property 
   * @param order 
   * @param filter 
   */
  listLocationsByFilters(page: number, size: number, property: String, order: String, filter: String): Observable<PageRequest>
  {
    return this.http.get(this.url + 'listLocationsByFilters/'+ page + '/' + size + '/' + property + '/' + order + '/' + filter).map(res => res.json());
  }
  
  /**
   * 
   * @param page 
   * @param size 
   * @param property 
   * @param order 
   */
  listLocations(page: number, size: number, property: String, order: String): Observable<PageRequest>
  {
    return this.http.get(this.url + 'listLocations/'+ page + '/' + size + '/' + property + '/' + order ).map(res => res.json());
  }



} 

File: projeto_construcao/src/main/java/br/com/projeto/domain/service/LocationService.java
package br.com.projeto.domain.service;

import java.util.List;

import org.directwebremoting.annotations.RemoteMethod;
import org.directwebremoting.annotations.RemoteProxy;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Sort.Direction;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.stereotype.Service;

import br.com.projeto.domain.entity.Location;
import br.com.projeto.domain.repository.ILocationRepository;

/**
 * 
 * @author André
 * @category Service
 * 
 */
@Service
@RemoteProxy(name = "locationService")
public class LocationService 
{
	
	/*-------------------------------------------------------------------
	 *				 		     ATTRIBUTES
	 *-------------------------------------------------------------------*/
	//Repository
	/**
	 * 
	 */
	@Autowired
	private ILocationRepository locationRepository;
	
	/*-------------------------------------------------------------------
	 *				 		     SERVICES
	 *-------------------------------------------------------------------*/
	
	/**
	 * 
	 * @param location
	 */
	@RemoteMethod
	public void insertLocation(Location location)
	{
		locationRepository.save(location);
	}
	
	/**
	 * 
	 * @return
	 */
	public List<Location> listAllLocation()
	{
		return locationRepository.findAll();
	}

	/**
	 * 
	 * @param id
	 * @return
	 */
	public List<Location> listAllSubLocation(Long id) 
	{
		return locationRepository.findAllSubLocations(id);
	}

	/**
	 * 
	 * @param id
	 * @return
	 */
	public Location findLocationById(Long id) 
	{
		return locationRepository.findOne(id);
	}
	

	/**
	 * 
	 * @param id
	 */
	@PreAuthorize("hasRole('ROLE_ADMIN')")
	public void deleteLocation(Long id) 
	{
		locationRepository.delete(id);
	}

	/**
	 * 
	 * @param location
	 */
	public void updateLocation(Location location) 
	{
//		if ( location.getLocation().getId() == location.getId() )
//		{
//			throw new IllegalArgumentException("Nâo foi possível salvar a localização.");
//		}
//		if ( (location.getLocation().getLocation().getLocation()).equals(null)  )
//		{
//			throw new IllegalArgumentException("A localização selecionada é uma localização principal");
//		}
		
//		if ( !( (location.getLocation()).equals(null) )  )
//		{
//			Long id = location.getId();
//			Long idMain = location.getLocation().getId();
//			List<Location> locations = locationRepository.findAllSubLocations(idMain);
//			Boolean existe = false;
//			for (int i=0; locations.size() < i; i++)
//			{
//				if (  )
//			}
//			throw new IllegalArgumentException("A localização selecionada é uma localização principal");
//		}
		locationRepository.saveAndFlush(location);
	}
    /**
     * 
     * @param page
     * @param size
     * @param property
     * @param order
     * @param filter
     * @return
     */
	public Page<Location> listLocationsByFilters(int page, int size, String property, String order, String filter) 
	{
		Direction asc;
		
		if (order.equals("ASC"))
		{
			asc = Direction.ASC;
		}
		else
		{
			asc = Direction.DESC;
		}
		PageRequest pageable = new PageRequest(page, size, asc, property);
	    System.out.println(pageable);
		return locationRepository.listLocationsByFilters(filter.toLowerCase(), pageable);
	}
    /**
     * 
     * @param page
     * @param size
     * @param property
     * @param order
     * @return
     */
	public Page<Location> listLocations(int page, int size, String property, String order) 
	{
		Direction asc;
		
		if (order.equals("ASC"))
		{
			asc = Direction.ASC;
		}
		else
		{
			asc = Direction.DESC;
		}
		
		PageRequest pageable = new PageRequest(page, size, asc, property);
	    System.out.println(pageable);
		return locationRepository.findAll(pageable);
	}
	
	

}
